pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
This has nothing to do with setup_requires - I've put that on the shelf while I prep more support that it will need - like automatic wheel building and caching.